pybossa/exporter/csv_reports_export.py

```diff
@@ -41,69 +40,60 @@ def download_name(self, project, ty):
         filename = secure_filename(filename)
         return filename
 
-    def _get_csv(self, out, writer):
-        out.seek(0)
-        yield out.read()
-
-    def _respond_csv(self, ty, id, info_only=False, **kwargs):
-        out = tempfile.TemporaryFile()
-        writer = UnicodeWriter(out)
-        empty_row = []
-        p = project_repo.get(id)
+    def _respond_csv(self, ty, project_id, info_only=False, **kwargs):
+        p = project_repo.get(project_id)
         if p is not None:
-            project_section = ['Project Statistics']
+            project_section = 'Project Statistics'
             project_header = ['Id', 'Name', 'Short Name', 'Total Tasks',
                               'First Task Submission', 'Last Task Submission',
                               'Average Time Spend Per Task', 'Task Redundancy']
-            writer.writerow(project_section)
-            writer.writerow(project_header)
-
 
             # get project data for report
             start_date = kwargs.get("start_date")
             end_date = kwargs.get("end_date")
-            project_data = get_project_report_projectdata(id, start_date, end_date)
-            writer.writerow(project_data)
+            project_data = get_project_report_projectdata(project_id, start_date, end_date)
 
-            writer.writerow(empty_row)
-            user_section = ['User Statistics']
+            project_csv = pd.DataFrame([project_data],
+                                       columns=project_header).to_csv(index=False)
+
+            user_section = 'User Statistics'
             user_header = ['Id', 'Name', 'Fullname', 'Email', 'Admin', 'Subadmin', 'Enabled', 'Languages',
                            'Locations', 'Start Time', 'End Time', 'Timezone', 'Type of User',
                            'Additional Comments', 'Total Tasks Completed', 'Percent Tasks Completed',
                            'First Task Submission', 'Last Task Submission', 'Average Time Per Task']
-            writer.writerow(user_section)
 
             # get user data for report
-            users_project_data = get_project_report_userdata(id, start_date, end_date)
-            if users_project_data:
-                writer.writerow(user_header)
-                for user_data in users_project_data:
-                    writer.writerow(user_data)
+            users_project_data = get_project_report_userdata(project_id, start_date, end_date)
+            if users_project_data and users_project_data[0]:
+                users_csv = pd.DataFrame(users_project_data, columns=user_header).to_csv(index=False)
             else:
-                writer.writerow(['No user data'])
+                users_csv = pd.DataFrame(columns=user_header).to_csv(index=False)
```

Comment:
Earlier report used to mention no user data which seems getting lost now?

Reply:
"Earlier report used to mention no user data" will never be hit because ```users_project_data``` is always true(because [[]] is True). When it enters the "if" condition, it writes the header, but didn't enter the "for" loop. And the way I changed follow this rule (with "else" condition, only writing the header).

Reply:
I understand this. Was wondering if the reports to look consistent as before; writing `No user data` instead of header as before?

Reply:
Yeah, it's consistent. Before, with no data, it writes a header with empty line (won't hit the 'else" block). So does my changes.